lib/ui/platform_dispatcher.dart

```diff
@@ -244,6 +248,28 @@ class PlatformDispatcher {
     _onMetricsChangedZone = Zone.current;
   }
 
+  FlutterView _createView(int id) {
+    return FlutterView._(id, this);
+  }
+
+  void _addView(int id) {
+    assert(id != _kImplicitViewId, 'The implicit view #$id can not be added.');
```

Comment:
This means that even if the implicit view is disabled, the embedder cannot create a view with ID 0. Is that a necessary limitation?

Reply:
Also see this comment: https://github.com/flutter/engine/blob/e6b8292705a832a56ab31b13fd819030070962c0/shell/platform/darwin/macos/framework/Source/FlutterView.h#L21

Reply:
This is my intended design. The idea is that, since the implicit view and the regular views have different models, some code might distinguish them by testing the view ID, since they might not have the signal on whether the id-0 view is an implicit one. It's much safer (and actually easier to understand) to just preserve ID 0 for the implicit view.

Reply:
I don't think we should give these IDs any semantic meaning like that publicly. These are just identifiers identifying a FlutterView, but the exact value doesn't have a meaning. If we come across a use case where it is important to distinguish between implicit view and regular views, we should add an explicit API for that (we already have this in the dart:ui world with PlatformDispatcher.implicitView).